### PR TITLE
CAT-1529 List related functions grouped together in FE

### PR DESCRIPTION
**Pull request by Fasusu**
Reordered functions and their parameters so that all list related functions are grouped together.
Also added new testclass for the list fragment of FE.

Run at Jenkins:
https://jenkins.catrob.at/view/All-Categories/view/Catroid/job/Catroid-CustomEmulator-Single-ClassAndPackage-Test/555/

### CODE REVIEW

**Comment by robertpainsi** [comment]
@FormulaEditorListFragmentTest.java

Always take a look at the [Android Studio code analysis](https://www.jetbrains.com/idea/help/code-analysis.html) result of your code.

- Use F2 to navigate throw the warnings.
- Use Alt+Enter to show the list of available [intention actions](https://www.jetbrains.com/idea/help/intention-actions.html) to resolve the warnings automatically.

Sometimes the available [intention actions](https://www.jetbrains.com/idea/help/intention-actions.html) don't fit and you have to resolve them manually.
e.g. "Rename 'item' to 'ignored'" at [diff](https://github.com/Catrobat/Catroid/pull/1386/files#diff-7507d10c4029f35dac85da12f8194fc7R115) doesn't make sense. The for-each loop should be replaced by a for loop.

After fixing all warnings, run the [Inspect Code tool](https://www.jetbrains.com/idea/help/running-inspections.html) over the file. The tool will show you another two warnings. Right click on each item and click on the yellow light bulb. This will automatically fix these warnings.

File: catroidTest/src/org/catrobat/catroid/uitest/formulaeditor/FormulaEditorListFragmentTest.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+		Script startScript1 = new StartScript();
+		Script startScript2 = new StartScript();
+		Brick changeBrick = new ChangeSizeByNBrick(0);
+		Brick glideToBrick = new GlideToBrick(0, 0, 0);
+
+		firstSprite.addScript(startScript1);
+		secondSprite.addScript(startScript2);
+		startScript1.addBrick(changeBrick);
+		startScript2.addBrick(glideToBrick);
```
Comment (robertpainsi) [inline catroidTest/src/org/catrobat/catroid/uitest/formulaeditor/FormulaEditorListFragmentTest.java]: Keep the code short, remove every code line which isn't necessary. Like bricks or the seconds sprite.

File: catroidTest/src/org/catrobat/catroid/uitest/formulaeditor/FormulaEditorListFragmentTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	}
+
+	public void testFunctionsListElements() {
+
+		solo.clickOnView(solo.getView(R.id.brick_change_size_by_edit_text));
+		assert(solo.getString(R.string.formula_editor_title).equals(solo.getCurrentActivity().getActionBar().getTitle()));
+
+		solo.clickOnText(solo.getString(R.string.formula_editor_math));
+		assertTrue(solo.getString(R.string.formula_editor_math), solo.waitForText(solo.getString(R.string.formula_editor_math)));
```
Comment (robertpainsi) [inline catroidTest/src/org/catrobat/catroid/uitest/formulaeditor/FormulaEditorListFragmentTest.java]: The solo and asserts seem to be a little odd. You can do better :wink:

Some solo methods have an internal timeout and therefore doesn't need to be checked. Take a look at http://robotium.googlecode.com/svn/doc/com/robotium/solo/Solo.html

File: catroidTest/src/org/catrobat/catroid/uitest/formulaeditor/FormulaEditorListFragmentTest.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+		int functionIndex = 0;
+
+		for(int item : FUNCTIONS_ITEMS) {
```
Comment (robertpainsi) [inline catroidTest/src/org/catrobat/catroid/uitest/formulaeditor/FormulaEditorListFragmentTest.java]: Missing whitespace after `for`.

**Comment by thmq** [comment]
@Fasusu: please squash your commits into one.

**Comment by thmq** [comment]
please refactor this test as a JUnit test.

**Comment by thmq** [comment]
please refactor according to our last conversation and re-open when done.

**Comment by Fasusu** [comment]
I rebased with current develop today and pushed to my repo but I can't reopen this pull request for some reason.

**Pull request closed without merging**